src/test/integration/ipv-stub/ipv-token.test.ts

```diff
@@ -0,0 +1,154 @@
+    expect(typeof tokenResponse.access_token).toBe("string");
+    expect(tokenResponse.token_type).toBe("Bearer");
+  });
+
+  it("should return 405 for GET request", async () => {
+    const response = await api.get("/token").expect("Content-Type", /json/);
+    expect(response.statusCode).toBe(405);
+    expect(response.body.message).toBe("Method GET not allowed");
+  });
+
+  it("should return 400 request with missing body", async () => {
+    const response = await api
+      .post("/token")
+      .set("Content-Type", "application/x-www-form-urlencoded")
+      .send();
+    expect(response.statusCode).toBe(400);
+    expect(response.body.message).toBe("Missing request body");
+  });
+
+  it("should return 400 request with unexpected grant type", async () => {
+    const response = await api
+      .post("/token")
+      .send(
+        await generateQuery(
+          "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+          "12345",
+          "invalid",
+          "https://api.identity/token",
+          "https://api.identity",
+          "authOrchestrator",
+          privateKeyValid
+        )
+      )
+      .expect("Content-Type", /json/);
+    expect(response.statusCode).toBe(400);
+  });
+
+  it("should return 400 request with unexpected client assertion type", async () => {
+    const response = await api
+      .post("/token")
+      .send(
+        await generateQuery(
+          "invalid",
+          "12345",
+          "authorization_code",
+          "https://api.identity/token",
+          "https://api.identity",
+          "authOrchestrator",
+          privateKeyValid
+        )
+      )
+      .expect("Content-Type", /json/);
+    expect(response.statusCode).toBe(400);
+  });
+
+  it("should return 400 request with missing auth code", async () => {
+    const response = await api
+      .post("/token")
+      .send(
+        await generateQuery(
+          "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+          "",
+          "authorization_code",
+          "https://api.identity/token",
+          "https://api.identity",
+          "authOrchestrator",
+          privateKeyValid
+        )
+      )
+      .expect("Content-Type", /json/);
+    expect(response.statusCode).toBe(400);
+  });
+
+  it("should return 500 request with invalid client assertion", async () => {
+    const response = await api
+      .post("/token")
+      .send(
+        await generateQuery(
+          "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+          "12345",
+          "authorization_code",
+          "https://api.identity/token",
+          "https://api.identity",
+          "authOrchestrator",
+          privateKeyInvalid
+        )
+      )
+      .expect("Content-Type", /json/);
+    expect(response.statusCode).toBe(500);
+  });
+});
+
+async function generateQuery(
+  clientAssertionType: string,
+  authCode: string,
+  grantType: string,
+  resource: string,
```

Comment:
We're actually removing this:
https://github.com/govuk-one-login/authentication-api/pull/5041

Reply:
Removed